sources/packages/api/src/services/student/student.service.ts

```diff
@@ -174,4 +173,39 @@ export class StudentService extends RecordDataModelService<Student> {
       this.repo.save(studentToUpdate);
     }
   }
+
+  /**
+   * Update the PD Sent Date
+   * @param studentId
+   */
+  async updatePDSentDate(studentId: number): Promise<Student> {
+    // get the Student Object
+    const studentToUpdate = await this.repo.findOne({
+      id: studentId,
+    });
+    if (studentToUpdate) {
+      studentToUpdate.StudentPDSentAt = new Date();
+      return this.repo.save(studentToUpdate);
+    }
+  }
+
+  /**
+   * Update the PD Sent Date
+   * @param studentId
+   * @param status
+   */
+  async updatePDStatusNDate(
+    studentId: number,
+    status: boolean,
+  ): Promise<Student> {
+    // get the Student Object
+    const studentToUpdate = await this.repo.findOne({
+      id: studentId,
+    });
+    if (studentToUpdate) {
```

Comment:
I would recommend throwing an exception if the student is not found, for instance calling findOneOrFail.
If the student is not found it will execute quietly, but it seems to be an error.